Update build lanes

loremattei opened the pull request:
This PR adds GitHub releases creation to the lanes that build the beta and the final artefacts.
It also tweaks the `code_freeze` lane to update the `release_notes.txt` file with the draft release notes so that they are available when building the first beta.

## To Test
**1. Code freeze lane**
You can simulate a complete code freeze, but since the work the new action does is unrelated to the other steps, it can be safely tested alone.

1.a Checkout a new test branch from this one.
1.b Comment out all the steps in `code_freeze` other than lines ~~65-68~~ 65-69. Commit the changes.
_1.c Add some content to `WordPress/metadata/release_notes_short.txt`and commit the changes._
1.d `bundle install`
1.e Run `bundle exec fastlane code_freeze` and verify that `WordPress/metadata/release_notes.txt` has been updated with the content of the main `RELEASE_NOTES.txt` related to the current version _and that `WordPress/metadata/release_notes_short.txt` has been cleaned up._
1.f Delete the test branch

**2. Alpha build**
2.a Checkout a new `release/<xx>` branch from this one.
2.b Edit `build.gradle` with a new alpha version and versionCode. Commit the changes.
2.c Run `bundle install`.
2.d Run `bundle exec fastlane build_alpha create_release:true` and verify that the a release is drafted on GitHub and that the release notes are added, the "Pre-release" option is checked and the .apk is uploaded.
2.e Delete the release draft.
2.f Delete the test branch.

**3. Beta build**
3.a Checkout a new `release/<xx>` branch from this one.
3.b Edit `build.gradle` with a new beta version and versionCode. Commit the changes.
3.c Run `bundle install`.
3.d Run `bundle exec fastlane build_beta create_release:true` and verify that the a release is drafted on GitHub and that the release notes are added, the "Pre-release" option is checked and the .apk is uploaded.
3.e Delete the release draft.
3.f Delete the test branch.

**4. Release build**
**Note: Following these instructions, you'll break the current release rollout. You can resume it in Play Store, but you have to be very careful when doing it and the related cleanup. Otherwise, you can comment out the `upload_build_to_play_store` action in the `build_and_upload_release` lane as it's not impacted by these changes anyway.**
4.a Checkout a new release/ branch from this one.
4.b Edit `build.gradle` with a final version and versionCode. Commit the changes.
4.c Run `bundle install`.
4.d Run `bundle exec fastlane build_and_upload_release create_release:true` and verify that the a release is drafted on GitHub and that the release notes are added and the .apk and .aab are uploaded.
4.e Delete the artifact from Google Play.
4.f Delete the draft release.
4.g Delete the test branch.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

oguzkocer commented:
@loremattei I am trying to wrap up encrypted logging, so I may not be able to review this until Friday. Please let me know if you need it reviewed earlier.

oguzkocer commented:
@loremattei I am trying to wrap up encrypted logging, so I may not be able to review this until Friday. Please let me know if you need it reviewed earlier.

loremattei commented:
Thank you for the review @oguzkocer!

I've added a private lane to clear `release_notes_short.txt`. I've written it to handle an array of files so that it will be easy to add more release related files that we may want to clear at code freeze in the future.

I've updated the instruction in the PR: new parts in Italic. 
I've also merged `develop` into this branch to fix the conflicts. 

I think this is ready for another look! Thank you!
